java/cdk/src/main/java/sleeper/cdk/stack/CompactionStack.java

```diff
@@ -288,10 +294,80 @@ private void lambdaToFindCompactionJobsThatShouldBeCreated(CoreStacks coreStacks
         instanceProperties.set(COMPACTION_JOB_CREATION_CLOUDWATCH_RULE, rule.getRuleName());
     }
 
-    private void ecsClusterForCompactionTasks(CoreStacks coreStacks,
-                                              IBucket jarsBucket,
-                                              LambdaCode taskCreatorJar,
-                                              Queue compactionJobsQueue) {
+    private void lambdaToCreateCompactionJobsBatchedViaSQS(
+            CoreStacks coreStacks, IBucket jarsBucket, LambdaCode jobCreatorJar, Queue compactionJobsQueue) {
+
+        // Function to create compaction jobs
+        Map<String, String> environmentVariables = Utils.createDefaultEnvironment(instanceProperties);
+
+        String triggerFunctionName = Utils.truncateTo64Characters(String.join("-", "sleeper",
+                instanceProperties.get(ID).toLowerCase(Locale.ROOT), "compaction-job-creation-trigger"));
+        String functionName = Utils.truncateTo64Characters(String.join("-", "sleeper",
+                instanceProperties.get(ID).toLowerCase(Locale.ROOT), "compaction-job-creation-handler"));
+
+        IFunction triggerFunction = jobCreatorJar.buildFunction(this, "CompactionJobsCreationTrigger", builder -> builder
+                .functionName(triggerFunctionName)
+                .description("Create batches of tables and send requests to create compaction jobs for those batches")
+                .runtime(JAVA_11)
+                .memorySize(256)
```

Comment:
This seems very low. It is a simple function but 1GB would be safer.

Reply:
I've raised an issue:
- https://github.com/gchq/sleeper/issues/1960